tests/benchq/compilation/test_graph_sim_mini.py

```diff
@@ -0,0 +1,180 @@
+################################################################################
+# © Copyright 2022-2023 Zapata Computing Inc.
+################################################################################
+import numpy as np
+import pytest
+import stim
+from orquestra.integrations.qiskit.conversions import import_from_qiskit
+from orquestra.quantum.circuits import CNOT, CZ, Circuit, H, I, S
+from qiskit import QuantumCircuit
+
+from benchq.compilation import pyliqtr_transpile_to_clifford_t
+from benchq.compilation.gate_stitching import get_icm
+from benchq.compilation.graph_sim_mini import get_vertices
+
+
+@pytest.mark.parametrize(
+    "circuit",
+    [
+        Circuit([I(0)]),
+        Circuit([H(0)]),
+        Circuit([S(0)]),
+        Circuit([H(0), S(0), H(0)]),
+        Circuit([H(0), S(0)]),
+        Circuit([S(0), H(0)]),
+        Circuit([S.dagger(0)]),
+        Circuit([H(2)]),
+        Circuit([CZ(0, 1), H(2)]),
+        Circuit([H(0), S(0), CNOT(0, 1), H(2)]),
+        Circuit([CNOT(0, 1), CNOT(1, 2)]),
+        Circuit(
+            [
+                H(0),
+                S(0),
+                H(1),
+                CZ(0, 1),
+                H(2),
+                CZ(1, 2),
+            ]
+        ),
+        Circuit(
+            [
+                H(0),
+                H(1),
+                H(3),
+                CZ(0, 3),
+                CZ(1, 4),
+                H(3),
+                H(4),
+                CZ(3, 4),
+            ]
+        ),
+    ],
+)
+def test_stabilizer_states_are_the_same_for_simple_circuits(circuit):
+
+    target_tableau = get_target_tableau(circuit)
+    vertices = get_vertices(circuit)
+    graph_tableau = get_stabilizer_tableau_from_vertices(vertices)
+
+    assert tableaus_correspond_to_same_state(graph_tableau, target_tableau)
+
+
+@pytest.mark.parametrize(
+    "filename",
+    [
+        "single_rotation.qasm",
+        "h_chain_circuit.qasm",
+    ],
+)
+def test_stabilizer_states_are_the_same_for_larger_circuits(filename):
+    # we want to repeat the experiment here since pyliqtr_transpile_to_clifford_t
+    # is a random process.
```

Comment:
Can't we seed it?

Reply:
In such case, wouldn't storing the compiled circuit be easier than running it multiple times?

Reply:
The compiled circuit for the case of the h_chian_from_qasm is quite large. It would probably add considerable size to the PR.